src/actions/socket_actions.js

```diff
@@ -0,0 +1,41 @@
+const sio = require('socket.io-client');
+const ActionTypes = require('../constants').ActionTypes;
+const PayloadSources = require('../constants').PayloadSources;
+const Dispatcher = require('../dispatcher');
+const SocketStore = require('../stores/socket_store');
+
+class SocketActions{
+    init(url){
+        var socket = sio(url);
+        socket.on("connect", this.onConnected);
+
+        Dispatcher.dispatch({
+            actionType: ActionTypes.SOCKET_INITIALIZED,
+            socket: socket
+        });
+        console.log("socket initialized");
+    }
+
+    onConnected(){
+        Dispatcher.dispatch({
+            actionType: ActionTypes.SOCKET_CONNECTED
+        });
+    }
+
+    auth(token){
+        var socket = SocketStore.getSocket()
+        socket.emit('auth', token, function(response){
+            if(response.Success){
+                Dispatcher.dispatch({
+                    actionType: ActionTypes.SOCKET_AUTHENTICATED
+                });
+            }else{
+                Dispatcher.sispatch({
```

Comment:
Now that we've resolved the stores question, I think we just need to fix this typo and we'll be good to merge :)